staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -325,3 +514,49 @@ func initStoreData(ctx context.Context, store storage.Interface) ([]interface{},
 	}
 	return created, nil
 }
+
+func verifyWatchEvents(t *testing.T, watcher watch.Interface, expectedWatchEvents []watch.Event, customVerificationFn func(t *testing.T, actualWatchEvents []watch.Event)) {
```

Comment:
We have a similar code for the watch cache tests. I am open to a refactor.

Reply:
Yes - we should definitely refactor.

Where I would like to get is that we effectively generalize the test exists in watchcache now, move it to pkg/storage/testing and run it for both implementations.